Export package version

justindsmith opened the pull request:
Exports the current package version from the root level. This allows
users of this module to read the version of the library that has been
loaded.

justindsmith commented:
The ocagent (#128) requests the core library version as part of it's payload. This PR exposes the core lib version in a cleaner manner so that it can be easily consumed by the exporters.

justindsmith commented:
> Fix the build?

Yeah, I'm working on the fix for the tests now.

justindsmith commented:
I've tried to solve this in a way that keeps the overhead minimal (still one location to update the current version, namely the package.json) but also keeps the `core` package unbound to a node environment (relying on filesystem reads or any module from the core nodejs library). I’m now copying the package.json file into `build` directory after compilation, which means the compile time, build time, and import time all have access to the package.json at the same relative path.

/cc @kjin I believe the node 6 failure is just environmental. It works locally for me on node 6. Can you rerun the that workflow to see if it fixes itself?

Here are the failing logs:
```
{ Error: Command failed: npm run test
sh: 1: nyc: not found
```

kjin commented on packages/opencensus-core/src/common/version.ts:
```diff
@@ -0,0 +1,19 @@
+ * limitations under the License.
+ */
+
+const pjson = require('../../package.json');
```
Let's use the [`pjson`](https://www.npmjs.com/package/pjson) module which I believe recursively descends directories until it hits a package.json file. That way we don't need to copy it into the `build` directory.

justindsmith replied:
My only concern with that is that `pjson` uses the `fs` node library, which means that `core` would be node-only and wouldn't work in (for example) a browser context. Is that okay? My understanding architecturally was that `core` should be environment agnostic, with other packages (`opencensus-nodejs`) that are environment specific.

If that's _not_ true, then yeah, something like pjson is a nice fit for this use case for sure and I could easily add that in.

kjin replied:
You're right. In that case, to avoid the file copy would it be possible to use a try-catch to descend either 2 or 3 directories, or hardcode `'../../../package.json'` instead? (I would prefer the former, but either works)

justindsmith commented:
@isaikevych I've resurrected this older PR to get it cleaned up and pushed in. The main goal was to export the `version` of the core library in a way that a consumer could easily read it. This was necessary for the OCAgent exporter to report the version of the library per those specs.

I've incorporated the feedback from @kjin above into the review, but also went ahead and updated the ocagent exporter code to use the new `version` instead of loading it from the package itself.

Let me know if you have any questions!

songy23 commented:
Should this PR be merged?
